src/main.zig

```diff
@@ -650,6 +650,44 @@ fn optionalStringEnvVar(arena: Allocator, name: []const u8) !?[]const u8 {
     }
 }
 
+// Read an open file descriptor to a new file in `dst_dir`. Return a file name
+// of the new file.
+fn copyToFile(
+    arena: Allocator,
+    in_fd: std.os.fd_t,
+    dst_dir: Compilation.Directory,
+) ![]const u8 {
+    // the section below is inspired by Compilation.tmpFilePath
+    const new_name = blk: {
+        const s = std.fs.path.sep_str;
+        const rand_int = std.crypto.random.int(u64);
+        if (dst_dir.path) |p| {
+            break :blk try std.fmt.allocPrint(arena, "{s}" ++ s ++ "tmp" ++ s ++ "{x}", .{ p, rand_int });
+        } else {
+            break :blk try std.fmt.allocPrint(arena, "tmp" ++ s ++ "{x}", .{rand_int});
+        }
+    };
+    try dst_dir.handle.makePath("tmp");
```

Comment:
Why is using `zig-cache/tmp/compiler` (next to the build.zig) not an option?

Reply:
It trying to use the local cache directory, if that's what you mean. Can you elaborate/

Reply:
Ah, I should have phrased this better. The code does not explain, why `tmp` in the `local cache directory` should not be used. Can you add a brief comment?

Reply:
Sorry, I still don't follow your comment. Can you re-phrase?

Perhaps I should move `Compilation.tmpFilePath` out of `Compilation` and re-use the function from everywhere with a `Dir` instead of `Compilation`.

Reply:
Apologies for my confusing phrasing.
As I understand you, I think yes. `fmt` also is wasting binary space (as of now), so this sounds like the right tradeoff.

If it is next to build.zig, then it sounds logical to reuse the function in Compilation.zig as the only difference is the used suffix (you could use an empty one at the call site).

**However**, if this api is intended to be called from anywhere (not only next to a build.zig), then a separate function sounds okay to me.

Reply:
Your comment just uncovered a bug in my implementation: I asked for an exclusive file, but there was no retry mechanism on collisions. However, it does not have to be exclusive.

I also understood your suggestion! Refactoring and will move to a single function.